pkg/cmd/list-images.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"sigs.k8s.io/yaml"
+
+	kubeonev1beta1 "k8c.io/kubeone/pkg/apis/kubeone/v1beta1"
+	"k8c.io/kubeone/pkg/templates/images"
+)
+
+type listImagesOpts struct {
+	ManifestFile string `longflag:"manifest" shortflag:"m"`
+}
+
+func listImagesCmd(rootFlags *pflag.FlagSet) *cobra.Command {
+	opts := &listImagesOpts{}
+
+	cmd := &cobra.Command{
+		Use:     "list-images",
+		Short:   "List images that will be used",
+		Example: `kubeone list-images -m mycluster.yaml -t terraformoutput.json`,
+		RunE: func(*cobra.Command, []string) error {
+			manifestFile, err := rootFlags.GetString(longFlagName(opts, "ManifestFile"))
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			opts.ManifestFile = manifestFile
+
+			return listImages(opts)
+		},
+	}
+
+	return cmd
+}
+
+func listImages(opts *listImagesOpts) error {
+	var imgopts []images.Opt
+
+	configBuf, err := os.ReadFile(opts.ManifestFile)
```

Comment:
We should add a comment briefly describing why we don't use the existing function for parsing the manifest. I guess the reason is that the existing function automatically validates the manifest, which we don't need here at all.

Reply:
👍 
Yep, you are exactly right. Existing mechanism to "load" the config would also validate it and basically I'm forced to ether write full config manually for just image-list, or eve worse to create actual instances with terraform :D